min_heap_binary_tree.rb

```diff
@@ -15,6 +15,27 @@ def insert(node)
     set_min_heap_state
   end
 
+  def delete(node)
+    replacement_node = tree[-1]
+    delete_index = tree.find_index(node)
+
+    delete_node(delete_index)
+    parent_child_check(replacement_node, delete_index)
+  end
+
+  def find(target_node)
+    byebug
+    if target_node.nil?
+      nil
+    else
+      target_index = tree.find_index(target_node)
+      return tree[target_index]
```

Comment:
Ruby always returns the last line of the method by default.  So you would not need this `return` here.

Does this code account for the case where the passed in target is not in the array?